sys/vm/vm_object.c

```diff
@@ -464,16 +464,22 @@ vm_object_allocate_anon(vm_pindex_t size, vm_object_t backing_object,
     struct ucred *cred, vm_size_t charge)
 {
 	vm_object_t handle, object;
+	int capflag = OBJ_HASCAP;
 
 	if (backing_object == NULL)
 		handle = NULL;
 	else if ((backing_object->flags & OBJ_ANON) != 0)
 		handle = backing_object->handle;
 	else
 		handle = backing_object;
+#if __has_feature(capabilities)
```

Comment:
Why use an ifdef when we unconditionally set OBJ_HASCAP otherwise?

Reply:
My thinking was that unlike other cases, this adds several instructions that always run to no purpose on a non-cheri system, but I'd waffled about adding the ifdef so I'll drop it.